src/util/address.rs

```diff
@@ -727,6 +727,36 @@ impl Address {
             (Network::Testnet, _) | (Network::Regtest, _) | (Network::Signet, _) => true
         }
     }
+
+    /// Returns true if the given pubkey is directly related to the address payload.
+    ///
+    /// This is determined by directly comparing the address payload with either the
+    /// hash of the given public key or the segwit redeem hash generated from the
+    /// given key. For taproot addresses, the supplied key is assumed to be tweaked
+    pub fn is_related_to_pubkey(&self, pubkey: &PublicKey) -> bool {
+        let pubkey_hash = pubkey.pubkey_hash();
+        let payload = self.payload_as_bytes();
+
+        (*pubkey_hash == *payload) || (pubkey.to_bytes()[1..] == *payload) || (*segwit_redeem_hash(&pubkey_hash) == *payload)
```

Comment:
```
(pubkey.to_bytes()[1..] == *payload)
```
This is effectively the Taproot check. I don't think its possible to end up with a `pubkey.to_bytes()` with a length of less than 1 through any sane usage.